lib/apiservers/engine/backends/container.go

```diff
@@ -530,24 +541,29 @@ func (c *Container) containerStart(name string, hostConfig *container.HostConfig
 	// TODO: We need a resolved ID from the name
 	stateChangeRes, err := client.Containers.StateChange(containers.NewStateChangeParams().WithHandle(h).WithState("RUNNING"))
 	if err != nil {
-		if _, ok := err.(*containers.StateChangeNotFound); ok {
-			return derr.NewRequestNotFoundError(fmt.Errorf("server error from portlayer"))
-		}
+		switch err := err.(type) {
 
-		// If we get here, most likely something went wrong with the port layer API server
+		case *containers.StateChangeNotFound:
+			return derr.NewRequestNotFoundError(fmt.Errorf("server error from portlayer : %s", err.Payload.Message))
 
-		return derr.NewErrorWithStatusCode(fmt.Errorf("Unknown error from the exec port layer"), http.StatusInternalServerError)
+		case *containers.StateChangeDefault:
+			return derr.NewRequestNotFoundError(fmt.Errorf("server error from portlayer : %s", err.Payload.Message))
+		}
 	}
 
 	h = stateChangeRes.Payload
 
 	// commit the handle; this will reconfigure and start the vm
 	_, err = client.Containers.Commit(containers.NewCommitParams().WithHandle(h))
 	if err != nil {
-		if _, ok := err.(*containers.CommitNotFound); ok {
-			return derr.NewRequestNotFoundError(fmt.Errorf("server error from portlayer"))
+		switch err := err.(type) {
+
+		case *containers.CommitNotFound:
+			return derr.NewRequestNotFoundError(fmt.Errorf("server error from portlayer : %s", err.Payload.Message))
+
+		case *containers.CommitDefault:
+			return derr.NewRequestNotFoundError(fmt.Errorf("server error from portlayer : %s", err.Payload.Message))
```

Comment:
The explicit repetition is there in case we want to handle individual errors with more docker like responses and it makes it easier to add more error types. As for the defaults. they are now fixed.